### ISSUE TITLE
Uncaught error when creating connections

### ISSUE THREAD

**Issue by nickminutello** [carlos8f/haredis#15]
```
/myapp/2.2.32/server/node_modules/haredis/node_modules/redis/index.js:487
            throw err;
                  ^
TypeError: Cannot call method 'removeListener' of undefined
    at RedisHAClient.onReady.designateSubSlave.self.server_info (/myapp/2.2.32/server/node_modules/haredis/index.js:152:19)
    at Array.forEach (native)
    at RedisHAClient.onReady (/myapp/2.2.32/server/node_modules/haredis/index.js:151:16)
    at _asyncMap (/myapp/2.2.32/server/node_modules/haredis/node_modules/async/lib/async.js:190:13)
    at async.forEach (/myapp/2.2.32/server/node_modules/haredis/node_modules/async/lib/async.js:82:20)
    at _asyncMap (/myapp/2.2.32/server/node_modules/haredis/node_modules/async/lib/async.js:184:9)
    at Object.doParallel [as map] (/myapp/2.2.32/server/node_modules/haredis/node_modules/async/lib/async.js:168:23)
    at Object.async.parallel (/myapp/2.2.32/server/node_modules/haredis/node_modules/async/lib/async.js:452:19)
    at RedisHAClient.designateSubSlave (/myapp/2.2.32/server/node_modules/haredis/index.js:425:9)
    at RedisHAClient.onReady (/myapp/2.2.32/server/node_modules/haredis/index.js:136:8)

```

This is happening on startup of my app.
On startup, the app creates 5 connections.
I'm using cluster, with 5 workers - which are all starting up at the same time - however the error also occurs when just 1 worker is restarting..
I'm using haredis 0.2.8 - which is using redis 0.7.2
There are 2 redis servers on 2 boxes (v2.4.17)

**Comment by carlos8f** [comment]
Hi,

thanks for the bug report, the cause is simple, I didn't account for the fact that the `node.client` property is only defined after DNS resolution of the redis-server host. Should be fixed with c6971bc3fbf984b83da1578dfd6faaca87958b5b, and v0.2.13 and v0.2.14 both have that fix. v0.2.14 doesn't try to resolve the host if it's an ip.

Cheers,
Carlos

**Comment by nickminutello** [comment]
Thanks!
